### DISCUSSION TITLE
Diablo.ini setting to allow hotkeys to use of potions from the inventory

### DISCUSSION THREAD

**Discussion by cgicoffee** [diasurgical/DevilutionX#1567] [Ideas]
First of all, you guys are amazing. Replaying Diablo on Switch might've been the best gaming experience I've had in the last couple of years. And the joypad-adapted control scheme is ingenious! You rock.

Now, talking about the joypad... If you press L or R it will use a Health or a Mana potion from the belt. This is very useful and works great. The only problem is - if like me you play as Rogue, unfortunately you won't be able to beat Diablo with joypad-only controls, because you'll quickly run out of potions in your belt. And using them with the inventory open (like we all do on PC) while battling Diablo is out of the question since any open menu captures input focus. So opening the inventory, selecting a Health/Mana potion in the inventory and closing it takes long enough for the character to die from Diablo's attacks. Even if you try touch controls, touching a potion in the inventory just picks it up, you still need to use it with the button. And it's quite a "UI challenge" trying to "tap" Diablo while tapping the potions and hitting the Y key to use them (which is still not a joypad-only option).

I'd like to suggest a new diablo.ini parameter "Hotkeys Use All Potions" which, when enabled, allows the game to continue using potions *from the inventory* if there are none in the belt. First, they will use belt potions, naturally, and then - from the inventory. I thought this though and think it's an useful and organic improvement for joypad-controlled clients. The order for use can be, just like in the belt, left to right, but also top to bottom. As for the Rejuvenation potions: it can first look for potions of the type requested (say, Health on L press) and only if those are missing, go for the RPs.

**Comment by AJenbo** [comment]
Something similar, but more generic is under consideration

**Reply by cgicoffee** [reply]
That's great news. Hope it makes it into Master someday.
